server/project/project.go

```diff
@@ -210,6 +210,22 @@ func getRemovedSources(oldProj, newProj *v1alpha1.AppProject) map[string]bool {
 	return removed
 }
 
+func uniqueGroupKinds(slice []metav1.GroupKind) []metav1.GroupKind {
```

Comment:
Took me a while to figure out this function.  Can we describe it as: modifiedGroupKinds, or xorGroupKinds() and comment that it returns a list of group kinds that are exclusively different from each list. Also, the method signature could be modified to take a left/right []metav1.GroupKind list and the append can happen in the function.

Reply:
good idea